lib/smb2/file.rb

```diff
@@ -58,23 +79,23 @@ def inspect
   # @param length [Fixnum,nil] number of bytes to read. If this is nil, read
   #   the whole file, i.e. {#size} bytes.
   # @return [String] full contents of the remote file
-  def read(offset: 0, length: nil)
+  def read(length = nil, offset: self.pos)
```

Comment:
I'm not a huge fan of this mix of positional arguments with defaults and keyword arguments.  Was this so that our API was closer to the Ruby File API?